test_programs/plonky2_prove_failure/signed_casts/Prover.toml

```diff
@@ -0,0 +1,3 @@
+x = [ -128, 127 ]
```

Comment:
These should be [ -128, -128 ]. A correct signed conversion should extend the sign bit, e.g. -128 as i8 is "10000000". When converted to i16, it would become "1111111110000000", which is -128 as i16. In principle, sign extension works by repeating the sign bit, e.g. the bit pattern "sxyzabcd" becomes "sssssssssxyzabcd". This preserves negative numbers in two's complement form correctly.

Reply:
Well yes, but the test is for conversion correctness, so what if someone on accident makes it so you would **always** extend the sign bit, meaning with negatives it works but for positives it doesn't.
Obviously that is a really silly mistake to make, but I think it would be beneficial to check positive number cast, just in case.

Reply:
I mistakenly assumed that you're checking whether x[0] as i16 == x[1], but you're testing x[0] with y[0] and z[0]. So, everything is correct, it was my mistake.